apt-hook/json-hook.cc

```diff
@@ -238,25 +242,31 @@ struct ESMContext {
 
 ESMContext get_esm_context() {
     CloudID cloud_id = get_cloud_id();
-    bool is_x = is_xenial();
+    ESMInfraSeries esm_infra_series = get_esm_infra_series();
 
     ESMContext ret;
     ret.context = "";
     ret.url = "https://ubuntu.com/pro";
 
-    if (cloud_id != AZURE && is_x) {
+    if (cloud_id != AZURE && esm_infra_series == XENIAL) {
```

Comment:
As written, because of the `else if` later, this won't populate the context and url on e.g. AWS Focal.

But I'll try to organize this into two levels to make it more organized